backend/src/exercise/patient-ticking.ts

```diff
@@ -86,43 +67,30 @@ function getDedicatedResources(
     state: ExerciseState,
     patient: Patient
 ): Catering {
+    const cateringTypes: Catering = {
+        notarzt: 0,
+        notSan: 0,
+        rettSan: 0,
+        san: 0,
+        gf: 0,
+        material: 0,
+    };
+
     if (!patient.isBeingTreated) {
```

Comment:
Your sentence doesn't make any sense to me.

I assume (solely based on where the comment is made) you mean that `isBeingTreated` only considers personnel. Therefore this function doesn't produce the expected return value if the patient is only treated by material.

Can't we just remove the `isBeingTreated` check here?

Reply:
> Your sentence doesn't make any sense to me.
> 
> I assume (solely based on where the comment is made) you mean that `isBeingTreated` only considers personnel. Therefore this function doesn't produce the expected return value if the patient is only treated by material.
> 
> Can't we just remove the `isBeingTreated` check here?

I would not remove the `isBeingTreated` check, if we remove this check we could discuss if the variable `isBeingTreated` should be removed, as it is just kind of a performance caching value.

Right now `isBeingTreated === false` means no personnel is treating. If the material calculation is set to after this check we will return that this patient has `zero` material associated with him, even if he does. 
This comes from the dependence that we know that it doesn't matter if a patient has material or not when there is no personnel treating him/her, this is an indirect dependence.
But this is a dependence that doesn't need to be there. The dependence of `isBeingTreated === false` when no personnel is treating a patient is there as this is the whole purpose of this variable (to tell if a patient has at least one personnel treating him/her), but `isBeingTreated` doesn't say right now anything about how much material a patient has assigned.
This whole thing is right now not important as a patient without personnel will change his/her state independently from material assigned to him/her.
With putting it before we remove this dependence, otherwise I would advise at least writing a comment or description.

Reply:
Removing the `isBeingTreated`-check here would fix the return value of the function. It would also not impact the performance of this function by a significant amount as in the case that `isBeingTreated` is false, `patient.assignedPersonnelIds` is an empty object.

Your current solution with moving the `material.length` stuff up is not very elegant and could lead to future confusion/bugs.

So my two proposals:
1. Remove the `isBeingTreated`-check in this function
    - Fixes `getDedicatedResources`
    - Less code
2. Replace `isBeingTreated` everywhere with `Patient.isTreatedByPersonnel = (patient)=> !isEmpty(patient.assignedPersonnelIds)` 
    - Better name
    - The calculation is very fast and should therefore be preferred over setting the variable each time it changes
    - Performance impact should be minimal.

1. Should be done in this pull request
2. Could maybe be done by the next team (easy task to get familiar with refactoring (state, writing a migration, ...))

Reply:
First proposal is good, can be renamed later, maybe putting a TODO to the variable `isBeingTreated` and pointing to your second proposal?

Reply:
> First proposal is good, can be renamed later, maybe putting a TODO to the variable `isBeingTreated` and pointing to your second proposal (to do that later)?

Done